exovetter/vetters.py

```diff
@@ -132,16 +119,122 @@ def run(self, tce, lightcurve):
 
 
+class Sweet(BaseVetter):
+    """Class to handle SWEET Vetter functionality.
+
+    Parameters
+    ----------
+    threshold_sigma : float
+        Threshold for comparing signal to transit period.
+
+    Attributes
+    ----------
+    tce, lc
+        Inputs to :meth:`run`.
+
+    result : dict
+        ``'amp'`` contains the best fit amplitude, its uncertainty, and
+        amplitude-to-uncertainty ratio for half-period, period, and
+        twice the period. ``'msg'`` contains warnings, if applicable.
+        Populated by :meth:`run`.
+
+    lsf : `~exovetter.utils.WqedLSF`
+        Least squares fit object, populated by :meth:`run`.
+
+    """
+    def __init__(self, threshold_sigma=3):
+        self.tce = None
+        self.lc = None
+        self.result = None
+        self.lsf = None
+        self.threshold_sigma = threshold_sigma
+
+    def run(self, tce, lightcurve):
+        self.tce = tce
+        self.lc = lightcurve
+        epoch = tce.get_epoch(getattr(exo_const, lightcurve.time_format))
+
+        # TODO: Do we want results to have unit? If so, what?
+        self.result, self.lsf = self._do_fit(
+            lightcurve.time * u.day, lightcurve.flux_quantity,
+            tce['period'], epoch, tce['duration'])
+
+    def plot(self):  # pragma: no cover
+        import matplotlib.pyplot as plt
+
+        phase = self.lsf.x
+        flux = self.lsf.y
+        best_fit = self.lsf.get_best_fit_model()
+
+        fig, axes = plt.subplots(2, 1, sharex=True)
+        plt.subplots_adjust(hspace=0.001)
+        axes[0].plot(phase, flux, 'k.')
+        axes[1].plot(phase, best_fit, 'r.')
+        axes[0].set_ylabel('Flux')
+        axes[1].set_xlabel('Phase')
+        axes[0].set_title(f'{self.tce.get("target_name", "Target")} '
+                          f'({self.tce.get("event_name", "Event")})')
+        plt.draw()
+
+        return fig
+
+    def _do_fit(self, time, flux, period_in, epoch, duration_in):
+
+        if len(time) != len(flux):
+            raise ValueError('time and flux length mismatch')
+
+        threshold_sigma = self.threshold_sigma
+
+        idx = np.isnan(time) | np.isnan(flux)
+        time = time[~idx]
+        flux = flux[~idx]
+
+        # Discard units because the following calculations do not understand
+        # Quantity.
+        time = time.to_value(u.day)
+        flux = flux.to_value()  # Just have to trust lightkurve on this one
+        period_days = period_in.to_value(u.day)
+        duration_days = duration_in.to_value(u.day)
+        epoch = epoch.to_value(u.day)
+
+        idx = mark_transit_cadences(time, period_days, epoch, duration_days)
+        flux = flux[~idx]
+
+        out = []
+        for per in [period_days * 0.5, period_days, 2 * period_days]:
+            phase = np.fmod(time - epoch + per, per)
+            phase = phase[~idx]
+            period = np.max(phase)
+            f_obj = WqedLSF(phase, flux, None, period=period)
```

Comment:
As you've implemented WqedLsf fits a sine curve to data with a zero mean. But you haven't subtracted off the mean flux from the star in this function before fitting.  Did I miss it? The data from lightkurve is likely to have a mean value around which the flux oscillates.

Reply:
I really don't know all the science behind the implementation. What I have was translated from to the best of my ability:

https://github.com/exoplanetvetting/DAVE/blob/test_DAVE/tessPipeline/sweet.py

https://github.com/exoplanetvetting/DAVE/blob/test_DAVE/fileio/kplrfits.py

https://github.com/exoplanetvetting/DAVE/blob/test_DAVE/misc/lsf.py

Given that DAVE has no test for this, I was unable to vet my ported code properly.

Can you please suggest a fix using the "Insert a suggestion" feature? Thanks.

Reply:
I'll have a look at it. I think the DAVE code assumes the mean has already been subtracted/